lib/base/networking/apis/auth_handler.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:gocast_mobile/routes.dart';
+
+class AuthHandler {
+  /* static Future<void> saveToken(List<Cookie> cookies) async {
+    for (var cookie in cookies) {
+      if (cookie.name == 'jwt') {
+        SharedPreferences prefs = await SharedPreferences.getInstance();
+        await prefs.setString('jwt', cookie.value);
+
+        // DEBUG: Check cookie
+        String? jwt = prefs.getString('jwt');
+        debugPrint('Current jwt: $jwt');
+        return;
+      }
+    }
+
+    // TODO: Handle error when no jwt cookie is found
+    debugPrint('No JWT cookie found');
+  }
+*/
+  static Future<void> basicAuth(
+    String username,
+    String password,
+  ) async {
+    const url = Routes.basicLogin;
+    final cookieJar = CookieJar();
+    final dio = Dio(
+      BaseOptions(
+        followRedirects: false,
+        validateStatus: (status) {
+          return status! < 500;
+        },
+      ),
+    );
+    dio.interceptors.add(CookieManager(cookieJar));
+
+    final formData = FormData.fromMap({
+      'username': username,
+      'password': password,
+    });
+
+    try {
+      await dio.post(
+        url,
+        data: formData,
+      );
+    } catch (e) {
+      // TODO: Handle network errors
+      debugPrint('Request error: $e');
+      return;
+    }
+
+    List<Cookie> cookies = await cookieJar.loadForRequest(Uri.parse(url));
+
+    // Save jwt token
+    await Token.saveToken(cookies);
+  }
+
+  static Future<void> ssoAuth(BuildContext context) async {
+    debugPrint('ssoAuth started');
+    // Redirect the user to the Shibboleth login page
+    debugPrint('Login URL: $Routes.ssoLogin');
+
+    // Open the login page in a web view
+    await Navigator.push(
+      context,
+      MaterialPageRoute(
+        builder: (context) => webview.InAppWebView(
+          initialUrlRequest:
+              webview.URLRequest(url: Uri.parse(Routes.ssoLogin)),
+          onLoadStop:
+              (webview.InAppWebViewController controller, Uri? url) async {
+            debugPrint('Page load stopped: $url');
+
+            final cookieManager = webview.CookieManager.instance();
+            List<webview.Cookie> cookies =
+                await cookieManager.getCookies(url: url!);
+
+            // Save jwt token
+            await Token.saveToken(
+              cookies.map((c) => Cookie(c.name, c.value)).toList(),
+            );
+
+            // Redirect back to app
+            if (url.toString().startsWith(Routes.ssoRedirect)) {
+              debugPrint('Redirect URL detected: $url');
+
+              // Close the web view and go back to the app
+              debugPrint('Closing web view and returning to app');
+              Navigator.pop(context);
+            }
+          },
+        ),
+      ),
+    );
+  }
+
+  // Generate user mock for testing the views until API/v2 is implemented
+  static Future<User> fetchUser() async {
+    // TODO: Add GET:/user endpoint in gocast API to fetch current user information
```

Comment:
Just to mention - we can always (and should) convert these TODOs into tasks 🙃

Reply:
I had planned to first talk with Moritz, as we first wanted to define what endpoints we actually need and how the api/v2 should be structured before creating individual issues. ([see related issue](https://github.com/TUM-Dev/gocast_mobile/issues/13)).